From: Simon McVittie <smcv@debian.org>
Date: Fri, 25 Sep 2020 16:12:01 +0100
Subject: Revert "testsuite: Don't use g_test_skip"

When versions of Meson with bug
https://github.com/mesonbuild/meson/issues/7515 are no longer widespread,
the Pango test suite should diagnose which tests are skipped again.
Versions 0.55.0 to 0.55.2 inclusive have the bug; version 0.55.3 is OK.

In Debian we know we have meson 0.55.3, so we can apply this early.

This reverts commit 440f9c2609b7f5f9d8f65619b029a5eb660de88b.

Forwarded: https://gitlab.gnome.org/GNOME/pango/-/merge_requests/246
Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tests/test-break.c         | 6 ------
 tests/test-itemize.c       | 3 ---
 tests/test-layout.c        | 3 ---
 tests/testboundaries_ucd.c | 2 --
 4 files changed, 14 deletions(-)

diff --git a/tests/test-break.c b/tests/test-break.c
index c435964..413fd08 100644
--- a/tests/test-break.c
+++ b/tests/test-break.c
@@ -72,12 +72,9 @@ test_file (const gchar *filename, GString *string)
 
   if (pango_layout_get_unknown_glyphs_count (layout) > 0)
     {
-#if 0
-      // See https://github.com/mesonbuild/meson/issues/7515
       char *msg = g_strdup_printf ("Missing glyphs - skipping %s. Maybe fonts are missing?", filename);
       g_test_skip (msg);
       g_free (msg);
-#endif
       g_free (contents);
       g_object_unref (layout);
       pango_attr_list_unref (attributes);
@@ -263,12 +260,9 @@ test_break (gconstpointer d)
   setlocale (LC_ALL, "en_US.utf8");
   if (strstr (setlocale (LC_ALL, NULL), "en_US") == NULL)
     {
-#if 0
-      // See https://github.com/mesonbuild/meson/issues/7515
       char *msg = g_strdup_printf ("Locale en_US.UTF-8 not available, skipping break %s", filename);
       g_test_skip (msg);
       g_free (msg);
-#endif
       g_free (old_locale);
       return;
     }
diff --git a/tests/test-itemize.c b/tests/test-itemize.c
index b22f3c2..dbe67e6 100644
--- a/tests/test-itemize.c
+++ b/tests/test-itemize.c
@@ -241,12 +241,9 @@ test_itemize (gconstpointer d)
   setlocale (LC_ALL, "en_US.utf8");
   if (strstr (setlocale (LC_ALL, NULL), "en_US") == NULL)
     {
-#if 0
-      // See https://github.com/mesonbuild/meson/issues/7515
       char *msg = g_strdup_printf ("Locale en_US.UTF-8 not available, skipping itemization %s", filename);
       g_test_skip (msg);
       g_free (msg);
-#endif
       return;
     }
 
diff --git a/tests/test-layout.c b/tests/test-layout.c
index a5b6d27..2ed2a73 100644
--- a/tests/test-layout.c
+++ b/tests/test-layout.c
@@ -311,12 +311,9 @@ test_layout (gconstpointer d)
   setlocale (LC_ALL, "en_US.utf8");
   if (strstr (setlocale (LC_ALL, NULL), "en_US") == NULL)
     {
-#if 0
-      // See https://github.com/mesonbuild/meson/issues/7515
       char *msg = g_strdup_printf ("Locale en_US.UTF-8 not available, skipping layout %s", filename);
       g_test_skip (msg);
       g_free (msg);
-#endif
       return;
     }
 
diff --git a/tests/testboundaries_ucd.c b/tests/testboundaries_ucd.c
index 1f0276e..18ab94c 100644
--- a/tests/testboundaries_ucd.c
+++ b/tests/testboundaries_ucd.c
@@ -232,9 +232,7 @@ do_test (const gchar *filename,
   channel = g_io_channel_new_file (filename, "r", &error);
   if (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
     {
-#if 0
       g_test_skip ("Test file not found");
-#endif
       return;
     }
 
